core/src/main/java/com/aem/demo/core/models/NewsCardModel.java

```diff
@@ -39,7 +44,62 @@ public class NewsCardModel {
     @Default(values = DEFAULT_IMAGE)
     private String image;
 
+//    private int likes;
+
+    @Inject
+    private SlingHttpServletRequest servletRequest;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+
     public String getPath() {
-        return resource.getPath().replace("/jcr:content","");
+        return resource.getPath().replace("/jcr:content", "");
+    }
+
+    public String getFullPath() {
+        return resource.getPath();
+    }
+
+
+    public int  getLikes() {
+        int likes = 0;
+
+        String rootPath = resource.getPath();
+
+        resource = resourceResolver.getResource(rootPath);
+        ModifiableValueMap properties = resource != null ? resource.adaptTo(ModifiableValueMap.class) : null;
+
+        if (properties != null) {
+            likes = resource.adaptTo(ValueMap.class).get("likes", 0);
```

Comment:
You have already retrieved a ValueMap for this resource in line 71 (a `ModifiableValueMap`, as it follows from its name, is a species of a `ValueMap`, Therefore, you don't need to create another one, just reuse the existing one